### PR TITLE
chore: updates Arc monitoring resources

### PR DESCRIPTION
**Pull request by nilekhc**
**Reason for Change**:

The Fluentd container is crashing on one of the customer's OpenShift clusters. Based on customer feedback, I am updating the resource values. I have manually tested this change on the ARO cluster.

**Requirements**

- [x] squashed commits
- [ ] included documentation
- [ ] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

**Does this change contain code from or inspired by another project?**

- [ ] Yes
- [x] No

**If "Yes," did you notify that project's maintainers and provide attribution?**

**Special Notes for Reviewers**:

### CODE REVIEW

File: manifest_staging/charts/csi-secrets-store-provider-azure/templates/arc-monitoring.yaml [hunk 1]
```diff
@@ -183,10 +183,10 @@ spec:
           resources:
             requests:
               cpu: 50m
-              memory: 100Mi
+              memory: 250Mi
```
Comment (aramase) [inline manifest_staging/charts/csi-secrets-store-provider-azure/templates/arc-monitoring.yaml:186]: Have these limits been tested? If yes, could you update the PR description on why this change is being made and a note about the testing?

File: manifest_staging/charts/csi-secrets-store-provider-azure/templates/arc-monitoring.yaml [hunk 2]
```diff
@@ -205,6 +205,10 @@ spec:
             - name: run-journal-vol
               mountPath: /run/log/journal
               readOnly: true
+          {{- if eq .Values.Azure.Cluster.Distribution "openshift" }}
+          securityContext:
+            privileged: true
```
Comment (aramase) [inline manifest_staging/charts/csi-secrets-store-provider-azure/templates/arc-monitoring.yaml:210]: The plugin also need `privileged: true` if running on openshift. I see we're not setting a conditional for arc here: https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/129931f8ed26453cb22c4dd903e4c1147ff8cab0/charts/csi-secrets-store-provider-azure/templates/provider-azure-installer.yaml#L88-L90

Is this being handled as part of the extension being enabled or we need to wrap this with a conditional `if eq .Values.Azure.Cluster.Distribution "openshift"` too?

File: manifest_staging/charts/csi-secrets-store-provider-azure/templates/arc-monitoring.yaml (hunk 1 above)
Comment (nilekhc) [reply manifest_staging/charts/csi-secrets-store-provider-azure/templates/arc-monitoring.yaml:186]: done

File: manifest_staging/charts/csi-secrets-store-provider-azure/templates/arc-monitoring.yaml (hunk 2 above)
Comment (nilekhc) [reply manifest_staging/charts/csi-secrets-store-provider-azure/templates/arc-monitoring.yaml:210]: We don't need `privileged: true` for the plugin. I could fetch the secret just fine without it on ARO. We are only setting `runAsUser: 0` in the `securityContext`

**Review by aramase — APPROVED** [review]
